data_encryption/models/encrypted_data.py

```diff
@@ -79,35 +80,33 @@ def _encrypted_read_json(self, name, env=None):
             return json.loads(data)
         except (ValueError, TypeError) as exc:
             raise ValidationError(
-                _("The data you are trying to read are not in a json format")
+                self.env._("The data you are trying to read are not in a json format")
             ) from exc
 
-    @staticmethod
-    def _retrieve_env():
+    def _retrieve_env(self):
         """Return the current environment
         Raise if none is found
         """
         current = config.get("running_env", False)
         if not current:
             raise ValidationError(
-                _(
+                self.env._(
                     "No environment found, please check your running_env "
                     "entry in your config file."
                 )
             )
         return current
 
-    @classmethod
-    def _get_cipher(cls, env):
+    def _get_cipher(self, env):
         """Return a cipher using the key of environment.
         force_env = name of the env key.
         Useful for encoding against one precise env
         """
-        key_name = "encryption_key_%s" % env
+        key_name = f"encryption_key_{env}"
         key_str = config.get(key_name)
         if not key_str:
             raise ValidationError(
-                _(
+                self.env._(
                     "No '%(key_name)s' entry found in config file. "
                     "Use a key similar to: %(key)s"
```

Comment:
how key_name is injected into this ?
where key is defined ?

Reply:
I am not sure to understand what you mean.
Key_name and key are injected 2 lines above

Reply:
1. I never know the "good way" ttto inject variable into strings, there are 10 ways to do it...
2. `key` is defined to lines below, when injecting the variable `"key": Fernet.generate_key()`

I can change the syntax with the `"my string".format() `one, is this what you suggest ?
`fstring` should not be used for translated string right ?

Reply:
1) yup.  you may try _("%(key_name)s", key_name=key_name) ` looks `_` accepts more than one arg https://www.odoo.com/documentation/18.0/developer/howtos/translations.html?highlight=translation#variables

2) Good !